nikola/plugins/template/jinja.py

```diff
@@ -112,7 +112,7 @@ def get_string_deps(self, text):
             filename = self.lookup.loader.get_source(self.lookup, dep_name)[1]
             deps.add(filename)
             sub_deps = self.get_deps(filename)
-            self.dependency_cache[dep_name] = sub_deps
+            self.dependency_cache[dep_name] = [filename] + sub_deps
```

Comment:
Would be simpler to just modify `sub_deps` right when we define it, wouldn’t it?